### PR TITLE
Log the test description if it is available

### PR DESCRIPTION
**Pull request by ktbarrett**
Fixes regression first reported [on gitter](https://gitter.im/cocotb/Lobby?at=61bf7839f20ff23319cc77a5) (ping @raysalemi). 

In cocotb 1.5 and prior, the test docstring was logged *verbatim* in the same message that contained the test start message. As Marlon pointed out, this was accidentally removed in #2507. This PR adds the message back, but handles the docstring a little smarter by normalizing the indentation using a [function provided in PEP257](https://www.python.org/dev/peps/pep-0257/#handling-docstring-indentation).

### CODE REVIEW

**Review by themperek — APPROVED** [review]

**Review by marlonjames — APPROVED** [review]

File: cocotb/regression.py [hunk 1]
```diff
@@ -472,19 +473,26 @@ def execute(self) -> None:
...
         if want_color_output():
             start = ANSI.COLOR_TEST
             end = ANSI.COLOR_DEFAULT
-        # Want this to stand out a little bit
+
+        if self._test.__doc__ is not None:
+            description = _trim(self._test.__doc__)
+        else:
+            description = ""
+
```
Comment (ktbarrett) [inline cocotb/regression.py:487]:
```suggestion
```

File: cocotb/regression.py [hunk 2]
```diff
@@ -472,19 +473,26 @@ def execute(self) -> None:
...
         self.log.info(
-            "{start}running{end} {name} ({i}/{total})".format(
+            "{start}running{end} {name} ({i}/{total}){description}".format(
                 start=start,
                 i=self.count,
                 total=self.ntests,
                 end=end,
                 name=self._test.__qualname__,
+                description=description,
```
Comment (ktbarrett) [inline cocotb/regression.py:495]: `_trim` already handles empty docstrings.
```suggestion
                description=_trim(self._test.__doc__),
```

File: cocotb/regression.py [hunk 3]
```diff
@@ -791,3 +799,35 @@ def generate_tests(self, prefix="", postfix=""):
...
             setattr(mod, name, _create_test(self.test_function, name, doc, mod,
                                             *self.args, **kwargs))
+
+
+def _trim(docstring: str) -> str:
+    """Normalizes test docstrings
+
+    Based on https://www.python.org/dev/peps/pep-0257/#handling-docstring-indentation.
+    """
+    if not docstring:
```
Comment (ktbarrett) [inline cocotb/regression.py:809]:
```suggestion
    if docstring is None:
```

File: cocotb/regression.py [hunk 4]
```diff
@@ -791,3 +799,35 @@ def generate_tests(self, prefix="", postfix=""):
                                "name_postfix" % (name, mod))
             setattr(mod, name, _create_test(self.test_function, name, doc, mod,
                                             *self.args, **kwargs))
+
+
+def _trim(docstring: str) -> str:
```
Comment (ktbarrett) [inline cocotb/regression.py:804]:
```suggestion
def _trim(docstring: Optional[str]) -> str:
```

**Review by marlonjames — APPROVED** [review]

**Pull request merged**